homeassistant/components/avreceiver/media_player.py

```diff
@@ -0,0 +1,201 @@
+"""Support for Denon AVR receivers using their HTTP interface."""
+from typing import Sequence
+
+from pyavreceiver.receiver import AVReceiver
+from pyavreceiver.zone import Zone
+
+from homeassistant.components.media_player import MediaPlayerEntity
+from homeassistant.components.media_player.const import (
+    DOMAIN,
+    SUPPORT_SELECT_SOUND_MODE,
+    SUPPORT_SELECT_SOURCE,
+    SUPPORT_TURN_OFF,
+    SUPPORT_TURN_ON,
+    SUPPORT_VOLUME_MUTE,
+    SUPPORT_VOLUME_SET,
+    SUPPORT_VOLUME_STEP,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+
+from .const import (
+    CONF_ZONE1,
+    DOMAIN as AVRECEIVER_DOMAIN,
+    SIGNAL_AVR_UPDATED,
+    THREE_DECIBELS,
+)
+
+BASE_SUPPORTED_FEATURES = (
+    SUPPORT_VOLUME_MUTE
+    | SUPPORT_VOLUME_SET
+    | SUPPORT_VOLUME_STEP
+    | SUPPORT_SELECT_SOURCE
+    | SUPPORT_TURN_ON
+    | SUPPORT_TURN_OFF
+)
+
+
+async def async_setup_entry(hass, entry: ConfigEntry, async_add_entities):
+    """Add media players for a config entry."""
+    zones = hass.data[AVRECEIVER_DOMAIN][entry.entry_id][DOMAIN]
+    entities = []
+    for name, zone in zones.items():
+        await zone.update_all()
+        if name == CONF_ZONE1:
+            entities.append(AVRMainZone(zone, name))
+        else:
+            entities.append(AVRZone(zone, name))
+    async_add_entities(entities)
```

Comment:
I will probably redo this in the dependency.  Just have a list of valid zones in the dependency so I don't have to do this fiddly stuff in __init__ and here